configure

```diff
@@ -0,0 +1,201 @@
+      exit 0
+      ;;
+    *)
+      echo "Unknown option: $arg"
+      exit 2
+      ;;
+    esac
+done
+
+vercomp () {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+check_go_version() {
+  echo -n "Checking go version... "
+  GO_VERSION=$(${tools[go]} version | ${tools[awk]} '{where = match($0, /[0-9]\.[0-9]+\.[0-9]*/); if (where != 0) print substr($0, RSTART, RLENGTH)}')
+  vercomp $GO_VERSION 1.10
+  case $? in
+    0) ;&
+    1)
+      printf "${GREEN}"
+      echo $GO_VERSION
+      printf "${NC}"
+      ;;
+    2)
+      printf "${RED}"
+      echo "$GO_VERSION < 1.10"
+      exit 1
+      ;;
+  esac
+}
+
+check_for() {
+  echo -n "Checking for $1... "
+  if ! [ -z "${desired[$1]}" ]; then
+    TOOL_PATH="${desired[$1]}"
+  else
+    TOOL_PATH=$(command -v $1)
+  fi
+  if ! [ -x "$TOOL_PATH" -a -f "$TOOL_PATH" ]; then
+    printf "${RED}not found${NC}\n"
+    cd - > /dev/null
+    exit 1
+  else
+    printf "${GREEN}found${NC}\n"
+    tools[$1]=$TOOL_PATH
+  fi
+}
+
+check_go_env() {
+  echo -n "Checking \$GOPATH... "
+  if [ -z "$GOPATH" ]; then
+    printf "${RED}invalid${NC} - GOPATH not set\n"
+    exit 1
+  fi
+  printf "${GREEN}valid${NC} - $GOPATH\n"
+}
+
+check_kubebuilder() {
+  echo -n "Checking for kubebuilder... "
+  if ! [ -z "${desired[kubebuilder]}" ]; then
+    TOOL_PATH="${desired[kubebuilder]}"
+  else
+    TOOL_PATH=$(command -v kubebuilder | sed s/\\/bin\\/kubebuilder$//)
+  fi
+  if ! [ -d "$TOOL_PATH" -a -f "$TOOL_PATH/bin/kubebuilder" -a -x "$TOOL_PATH/bin/kubebuilder" ]; then
+    printf "${RED}not found${NC}\n"
+    cd - > /dev/null
+    exit 1
+  else
+    printf "${GREEN}found${NC} - using $TOOL_PATH\n"
+    tools[kubebuilder]=$TOOL_PATH
+  fi
+}
+
+check_kubebuilder_version() {
+  echo -n "Checking kubebuilder version... "
+  ver=$( ${tools[kubebuilder]}/bin/kubebuilder version | awk '{where = match($0, /KubeBuilderVersion:"[0-9]\.[0-9]+\.[0-9]+"/); if (where != 0) print substr($0, RSTART+20, RLENGTH)}' | sed s/\",.\*// )
+  vercomp $ver 1.0.8
+  case $? in
+    0) ;&
+    1)
+      printf "${GREEN}"
+      echo $ver
+      printf "${NC}"
+      ;;
+    2)
+      printf "${RED}"
+      echo "$ver < 1.0.8"
+      exit 1
+      ;;
+  esac
+}
+
+cd ${0%/*}
+
+check_for make
+check_for awk
+check_for sed
+check_for shasum
+check_for tar
+check_for go
+check_go_version
+check_go_env
+
+if ! [[ ci -eq true ]]; then
```

Comment:
Why do we not perform these checks in CI? Does this not mean the `.env` file will have empty entries?